### PR TITLE
Describe AWS Secrets Storage backend

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Describe AWS Secrets Storage backend
- Add very short introduction storage backends 

## Testing

**WARNING:**  Make sure you installed use manifests from https://github.com/capactio/hub-manifests/pull/61

You don't need to install Mattermost:

1. Create and run Action to test the new default storage:

    1. Create TypeInstances:
        
        ```bash
        cat > /tmp/download-ti.yaml << ENDOFFILE
        typeInstances:
        - alias: "download"
          typeRef:
            path: cap.type.capactio.capact.validation.download
            revision: 0.1.0
          value:
            key: "true"
        ENDOFFILE
        export DOWNLOAD_TI=$(capact ti create -f /tmp/download-ti.yaml -ojson | jq -r '.[] | select(.alias == "download") | .id')
        cat > /tmp/update-ti.yaml << ENDOFFILE
        typeInstances:
        - alias: "update"
          typeRef:
            path: cap.type.capactio.capact.validation.update
            revision: 0.1.0
          value:
            key: "true"
        ENDOFFILE
        export UPDATE_TI=$(capact ti create -f /tmp/update-ti.yaml -ojson | jq -r '.[] | select(.alias == "update") | .id')
        ```

    2. Create Action input:
        ```bash
        cat > /tmp/act-input-ti.yaml << ENDOFFILE
        typeInstances:
          - name: "testUpdate"
            id: ${UPDATE_TI}
          - name: "testInput"
            id: ${DOWNLOAD_TI}
        ENDOFFILE
        ```
    3. Run Action
        ```bash
        capact act create cap.interface.capactio.capact.validation.action.passing --name test --type-instances-from-file /tmp/act-input-ti.yaml
        ```
    4. Run `capact act run test` and `capact act watch test`
    5. Get Action output TypeInstances:

        ```bash
        capact act get test -ojson | jq '.Actions[0].output.typeInstances'
        ``` 
    
       Observe the Backend ID near the `cap.type.capactio.capact.validation.upload` TypeInstance.

    6. Use `capact ti get {id} -oyaml` to see the details.

       ```bash
       SAMPLE_TI_ID=$(capact act get test -ojson | jq '.Actions[0].output.typeInstances | map(select(.typeRef.path == "cap.type.capactio.capact.validation.upload"))[0].id' -r)
       capact ti get $SAMPLE_TI_ID -oyaml
       ```

    7. See the AWS Secrets Manager UI to double confirm the TypeInstance value has been stored externally.
    8. Clean up the secret from AWS UI (as the delete functionality is not implemented yet).

## Related issue(s)

Resolves https://github.com/capactio/capact/issues/647

### CODE REVIEW

File: docs/feature/storage-backends/aws-secrets-manager.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+1. Follow the [AWS Credentials TypeInstance creation](../../example/typeinstances.md#aws-credentials) guide to create and obtain ID of the newly created TypeInstance.
+
+  Export it as `TI_ID` environment variable:
+
+  ```bash
+  export TI_ID="{id}"
+  ```
+
+1. Update Capact Global policy to inject the AWS credentials when necessary:
```
Comment (mkuziemko) [inline docs/feature/storage-backends/aws-secrets-manager.md:49]: we can add link to Global Policy

File: docs/feature/storage-backends/aws-secrets-manager.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+         description: "AWS Secrets Manager"
+ENDOFFILE
+capact policy apply -f /tmp/aws-storage-policy.yaml
+```
+
+To read more about TypeInstance backend configuration, see the [Definition of rules for TypeInstance](../policies/overview.md#definition-of-rules-for-typeinstance) section.
+
+## Next steps
+
+Now, you can run any Capact manifest and utilize the newly installed AWS Secrets Manager storage backend. For example, [install Mattermost](../../example/mattermost-installation.md). Once it's installed, verify the output TypeInstance values are created in AWS Secrets Manager. You can do it in two ways:
```
Comment (mkuziemko) [inline docs/feature/storage-backends/aws-secrets-manager.md:155]:
```suggestion
Now, you can run any Capact manifest and utilize the newly installed AWS Secrets Manager storage backend. For example, [install Mattermost](../../example/mattermost-installation.md). Once it's installed, verify the output TypeInstance values are created in AWS Secrets Manager. 
You can do it in two ways:
```

File: docs/feature/storage-backends/introduction.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Each storage backend is installed separately using Capact Actions. See corresponding documents for each storage to learn how to install and configure them.
+
+## Development
+
+To develop your own storage backend, create a gRPC service which implements the [storage backend Protocol Buffers schema](https://github.com/capactio/capact/blob/main/hub-js/proto/storage_backend.proto).
+
+You can use the generated server code for the following languages:
+- [Go](https://github.com/capactio/capact/tree/main/pkg/hub/api/grpc/storage_backend),
+- [TypeScript (Node.js)](https://github.com/capactio/capact/tree/main/hub-js/src/generated/grpc),
+- or, generate your own.
```
Comment (mkuziemko) [inline docs/feature/storage-backends/introduction.md:31]: I am not sure about this one but when I first read this page the comma looks strange. If it is intended then fine :) 

```suggestion
- or generate your own.
```

File: docs/feature/storage-backends/aws-secrets-manager.md (hunk 2 above)
Comment (pkosiec) [reply docs/feature/storage-backends/aws-secrets-manager.md:155]: I don't want to do the line break as is the logical continuation of previous sentence (you can do **it**). But I tried to tweak it a little 🙂

**Review by mkuziemko — APPROVED** [review]
LGTM

**Pull request merged**